.eslintrc.js

```diff
@@ -57,7 +56,16 @@ module.exports = {
             prefix: ['I'],
           },
         ],
+
+        // Disable ESLint core rules for which @typescript-eslint provides TypeScript-specific equivalents.
```

Comment:
Just curious, what's the harm in keeping them? if TS flags them first, then fixing the issue would also fix ESLint issue, would it not? (basically, I prefer fewer deviations from defaults if possible)

Reply:
The problem is that these can cause false positives with the new version of `@typescript-eslint`. E.g. removing the suppression for core's `no-shadow` rules results in a plethora of bogus reports about TS type definitions. The recommendation of silencing the corresponding core rules comes from the [plugin itself](https://typescript-eslint.io/rules/no-shadow/).

Reply:
Another way to fix it could be to make the root eslintrc also extend the `recommended` config from `@typescript-eslint`, because that config already does this toggling for us. This is what plexus does already. I'm thinking this could be done in a followup PR as it might cause some new errors.

Reply:
+1 -- #1258